docs/docs/guides/smart_contracts/Erc-20_transfer_with_EIP_1559.md

```diff
@@ -0,0 +1,121 @@
+
+# ERC-20 Transfers and EIP-1559
+
+## Introduction
+In this tutorial we will walk through sending transactions with the tx object and understand how the transaction gets executed, using web3.js version 4x.
+
+## The need for EIP-1559
+ 
+Before Users had to set a single gas price for their transactions. This led to high volatility in gas fees and issues with transactions getting stuck if the gas price wasn't high enough.
+With the introduction of EIP-1559, a minimum fee was introduced which will  automatically adjust based on network congestion
+also a tip for miners was introduced to incentivize them, which will prioritize the transaction.
+
+
+##  Overview
+
+Here is a high-level overview of the steps we will be taking in this tutorial:
+
+1. Setting up  
+2. Get current base fee from the network  
+3. Set max priority fee.  
+4. Construct the transaction object
+5. Sign and send the transaction to the RPC provider
+
+Before we start writing and deploying our contract, we need to set up our environment. For that, we need to install the following:
+
+1. Node.js - Node.js is a JavaScript runtime environment that allows you to run JavaScript on the server-side. You can download it from https://nodejs.org/en/download/  
+
+2. npm - Node Package Manager is used to publish and install packages to and from the public npm registry or a private npm registry. Here is how to install it https://docs.npmjs.com/downloading-and-installing-node-js-and-npm. (Alternatively, you can use yarn instead of npm https://classic.yarnpkg.com/lang/en/docs/getting-started/)
+
+# Getting started
+
+Here we initialize Web3 and set an RPC url inside `const Url;` , add our private key in the `const privateKey;` , then we specify the erc20 token inside the  
+ `const tokenAdress;`
+``` 
+Javascript
+
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract myContract{
+
+const Web3 = require('web3');
+
+
+const Url = 'wss://mainnet.infura.io/v3/******';
+
+const privateKey = 'YOUR_PRIVATE_KEY';
+
+const tokenAddress = '0x...'; 
+}
+
+```
+
+Inside the `async function transferWithEIP1559()`, we initialize a provider, then add account and get the current base fee from the network to store it inside of  
+`const gasPrice;` , next we set the maximum priority fee we are willing to allocate inside  `const maxPriorityFeePerGas;`, to do this we make use of utils package of web3.js, which is imported at the top.
+
+Construct the Transaction object:  
+To send the transaction details to the RPC provider for the transfer, we need to format the details in the tx object(transaction object). The tx object must contain the following:  
+from, to, value, gas, maxPriorityFeePerGas.
+
+
+```
+async function transferWithEIP1559() {
```

Comment:
code for `transferWithEIP1559` function should be complete in one snippet, as there is try block without catch in this code block, instead of multiple chunks

Reply:
do i move the explaination to the top or bottom?